fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+
+However, it does not contain a formalized way to filter based on arbitrary fields in an Item. For example, there is 
+no way to express the filter "item.properties.eo:cloud_cover is less than 10". It also does not have a way to logically combine
+multiple spatial or temporal filters.
+
+## Filter expressiveness 
+
+This extension expands the capabilities of Item Search and the OAFeat Items resource with 
+[OAFeat Part 3 CQL](https://portal.ogc.org/files/96288) 
+by providing an expressive query language to construct more complex filter predicates. The operators are similar to 
+those provided by SQL. The Simple CQL conformance class requires the logical operators `AND`, `OR`, and `NOT`; 
+the comparison operators '=', `<`, `<=`, `>`, `>=`, `LIKE`, `IS NULL`, `BETWEEN`, and `IN`; the spatial operator 
+`INTERSECTS`; and the temporal operator `ANYINTERACTS`. 
+
+The ANYINTERACTS operator has effectively the same semantics as the existing `datetime` filter.
+
+CQL enables these types of queries:
+- Use of Item Property values in predicates (e.g., `item.properties.eo:cloud_cover`), using comparison operators
+- Items whose `datetime` values are in the month of August of the years 2017-2021, using OR and ANYINTERACTS
+- Items whose `geometry` values intersect any one of several Polygons, using OR and INTERSECTS
+- Items whose `geometry` values intersect one Polygon, but do not intersect another one, using AND, NOT, and
+  INTERSECTS
+
+This extension also supports the Queryables mechanism that allows discovery of what Item fields can be used in 
+predicates.
+
+## OAFeat Part 3 Conformance Classes
+
+OAFeat CQL defines several conformance classes that allow implementers to create arbitrary compositions of 
+functionality that support whatever expressiveness they need.  Implementers many choose not to incur the cost of 
+implementing functionality they do not need, or may not be able to implement functionality that is not supported by their underlying datastore.  For example, Elasticsearch does not support the spatial predicates required by the Enhanced Spatial Operators conformance class.
+
+The Filter extension **requires** support of these three conformance classes:
+
+- Filter (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/filter`) - defines the Queryables mechanism and 
+  parameters `filter-lang`, filter-crs`, and `filter`
+- Features Filter (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/features-filter`) - defines that the parameters defined in `Filter` apply to the Features endpoint (`/collections/{collectionId}/items`) defined by OAFeat Part 1. 
+- Simple CQL (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/simple-cql`) - defines the query language used for the `filter` parameter defined by Filter
+
+This STAC Filter extension extends the Filter conformance class such that these parameters also apply
+to the STAC Item Search resource (/search). The OAFeat Filter conformance class already requires that these
```

Comment:
I realize now that I wrote this up wrong. Filter and Simple CQL are required, Features Filter is recommended, and Filter Extension is what binds Filter and Simple CQL to Item Search, in the same way Features Filter does to the Features endpoint /collections/{cid}/items

Reply:
But, I get what you're saying about the conformance classes and how STAC extends them.  I'm going to need to think about this some more.

Reply:
I thought about this more, and we need our own conformance classes that replicate those defined in Part 3. The primary reason is that the Features resource and /search would have to support exactly the same Part 3 classes if we reuse them, and that may not be the same. This also frees us up to change the semantics if we with, so we could, for example, say "our Filter conformance class allows use of properties not defined in Queryables".

I also change the format of these conformance classes to be /extension/filter rather than /item-search#filter -- I feel the former looks better and reduces coupling to to the other conformance classes.

Reply:
On further though, I changed them back to be like /item-search#filter style.